sapi/cgi/cgi_main.c

```diff
@@ -2264,6 +2257,13 @@ consult the installation file that came with this distribution, or visit \n\
 		}
 #endif
 		while (!fastcgi || fcgi_accept_request(request) >= 0) {
+#ifdef HAVE_VALGRIND
+			if (benchmark) {
+				/* measure startup and each benchmark run separately */
+				CALLGRIND_DUMP_STATS;
```

Comment:
I did many experiments and it seems Callgrind is not working correctly with php-src.

`CALLGRIND_DUMP_STATS` (https://valgrind.org/docs/manual/cl-manual.html#cr.dump-stats) should dump the collected data and zero counters. However, it seems the resulting data are wrong.

Here is an example of `Ir` counters collected for startup, 10 runs, and shutdown for non-JITted Symfony benchmark:
```
3790457 <-- startup
21834039 <-- first run repeat
861165355
39595343
39539066
39538301
39533537
39538461
39537382
39534526
39536394 <-- last run repeat
39535455 <-- shutdown
```

Notice, the 1st, 2nd, 3rd run times are very different. I would expect the 1st run to be the slowest. Also, shutdown does definitely not take about the same amount like the last run.

For JITted Symfony benchmark, the run counters are not even "stable across later runs":
```
3791102
23354110
1080202556
119984402
91536974
82315838
47140943
38638632
37998337
53000263
48679540
51503131
```

Based on these data it seems Callgrind is not working properly and some fix of php-src is required. I also tried /wo opcache, but the results are about the same as /w opcache.